File: lib/Entities/Order.dart
import 'dart:ffi';

class Order {
  final String walletId;
  final String type;
  final String asset;
  final String amount;
  final String price;
  final String stopPrice;
  final String status;
  final String action;

  Order(this.walletId, this.type, this.asset, this.amount, this.price, this.stopPrice, this.status, this.action);

  @override
  String toString() {
    return 'Wallet{imageUrl: $walletId, name: $type, hourlyPrice: $asset, prevPrice: $amount, price: $price, stopPrice: $stopPrice, action: $action}';
  }
}

File: FlutterNode/models/order.js
const mongoose = require('mongoose');

const orderSchema = new mongoose.Schema({
  walletId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Wallet',
    required: true
  },
  type: {
    type: String,
    required: true,
    enum: ['Market', 'Stop-Loss', 'Limit']
  },
  asset: {
    type: String,
    required: true,
    enum: ['BTC', 'ETH', 'LTC']
  },
  amount: {
    type: Number,
    required: true,
    min: 0
  },
  price: { // Target price for limit orders
    type: Number,
    required: function() {
      return this.type !== 'market';
    }
  },
  stopPrice: { // Target price for stop-loss orders
    type: Number,
    required: function() {
      return this.type === 'stop-loss';
    }
  },
  status: {
    type: String,
    enum: ['pending', 'completed', 'cancelled'],
    default: 'pending'
  },
  action: {
    type: String,
    required: true,
    enum: ['buy', 'sell']
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  completedAt: Date
}, { timestamps: true });

const Order = mongoose.model('Order', orderSchema);
module.exports = Order;
